### DISCUSSION TITLE
Error: export const dynamic = "force-dynamic" on page "/api/..." cannot be used with "output: export"

### DISCUSSION THREAD

**Discussion by mengxi-ream** [vercel/next.js#55937] [App Router]
I wrote some code in app router `app/api/generate` 
```
import { NextResponse, NextRequest } from "next/server";
// export const runtime = "edge"; // 'nodejs' is the default
export const dynamic = "force-dynamic"; // no caching

export async function POST(req: NextRequest) {
  try {
    return NextResponse.json({ success: true }, { status: 200 });
  } catch (error) {
    return error;
  }
}
```

And in config:
```
/** @type {import('next').NextConfig} */
const nextConfig = {
  output: "export",
  experimental: {
    swcPlugins: [["@swc-jotai/react-refresh", {}]],
  },
};

module.exports = nextConfig;
```
I got the error when I run `next build`, which is `Error: export const dynamic = "force-dynamic" on page "/api/generate" cannot be used with "output: export"`

**Comment by icyJoseph** [comment]
Hi,

`export` mode outputs a static HTML build. You can think of it as just JS, HTML and CSS, all of which can be put on a static files server, and be sent to a browser. If you have used CRA, then you can think of it, as _almost_, not quite, the same. The same part, is that CRA also outputs a static HTML build.

To have API routes or any sort of server work you need a server running your next application. And for that you can't have export mode.

**Reply by Tongre001** [reply]
So what's the solution here?

**Reply by icyJoseph** [reply]
Because this is a POST request... Making a build for export mode is not really possible. What is it you are trying to do?

Although, there's one way to deploy to these separately, but you'd still need a server.
